src/main/java/GameController.java

```diff
@@ -0,0 +1,47 @@
+import java.util.*;
+
+public class GameController {
+    //does all the fighting between Survivors and zombies
+    public static void fightRound(ArrayList<ICharacter> attackers, ArrayList<ICharacter> defenders) {
+        
+        for(int i = 0; i < attackers.size(); i++) {
+            for(int j = 0; j < defenders.size(); j++) {
+            
+                attackers.get(i).attack(defenders.get(j));
+
+                if(!defenders.get(j).isAlive()) 
+                    defenders.remove(j);
+                
+                if(defenders.isEmpty())
+                    break;
+            }
+        }
+    }
+    
+    public static void main(String[] args) {
+        //Spawns the Survivors and Zombies for the game
+        ArrayList<ICharacter> survivors = Spawner.spawnRandomSurvivors();
+        ArrayList<ICharacter> zombies = Spawner.spawnRandomZombies();
+
+        System.out.println("We have " + survivors.size() + " survivors trying to make it to safety.");
+        System.out.println("But there are " + zombies.size() + " zombies waiting for them.");
+
+        //keep fighting back and forth til either Survivors or Zombies are dead
+        while(survivors.size() > 0 || zombies.size() > 0) {
```

Comment:
This while loop condition should be an AND ( && ) not an OR ( | | ). "While there are more than 0 survivors AND more than 0 zombies"